app/controllers/transactions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class TransactionsController < ApplicationController
+  def new
+    @transaction = current_user.transactions.new
+  end
+
+  def create
+    @transaction = current_user.transactions.new(transaction_params)
+    if @transaction.save
+      redirect_to root_path, notice: t('.success')
+    else
+      render 'new', notice: t('.failure')
+    end
+  end
+
+  private
+
+  def transaction_params
+    params.require(:transaction).permit(:sum, :date, :comment)
+  end
+end
```

Comment:
@sinpai I didn't find test for this controller in `spec/controllers`.
Let's write request tests for both `new` and `create` methods

Reply:
👍 Will implement this. Thanks